src/ParseObject.js

```diff
@@ -1701,7 +1701,7 @@ class ParseObject {
       throw new Error('Cannot create an object without a className');
     }
     const constructor = classMap[json.className];
-    const o = constructor ? new constructor() : new ParseObject(json.className);
+    const o = constructor ? new constructor(json) : new ParseObject(json.className, Object.assign(json, { className: undefined }));
```

Comment:
`const o = constructor ? new constructor(json) : new ParseObject(json.className)` should be fine for your test to work.

There are other issues this brings up. There are other Subclasses of ParseObject like `_User` and `_Session` that will try to set readonly fields. You could try to clone the json and delete the readonly fields before passing to the constructor but I don't know what other issue this would cause.

Reply:
@RaschidJFR Thoughts?

Reply:
Yes, sorry I didn’t reply earlier. I’ve already done that with good results. Haven’t  pushed though.

 I’ve got almost all tests passing now, just a couple left.  What I’m trying to figure out now is a way to prevent pending ops from overwriting the properties set in the constructor. 

Tomorrow I’ll push my progress so anyone can have a look.